src/AzureDevOpsKubeService.ts

```diff
@@ -17,17 +17,35 @@ export class AzureDevOpsKubeService extends KubeServiceBase {
     fetch(resourceType: KubeResourceType): Promise<any> {
         switch (resourceType) {
             case KubeResourceType.Pods:
-                return Promise.resolve(JSON.parse(this._kubeSummaryData.pods));
+                return resolvePromise(this._kubeSummaryData.pods);

```

Comment:
Can this be `Promise.resolve(safeParseJson(this._kubeSummaryData.pods));` ?

Reply:
i want to avoid writing Promise.resolve(safeParseJson()) too many times, so did this way.

Reply:
I tend to agree with rajendra here as it isn't evident from `resolvePromise(this._kubeSummaryData.pods)` that we are actually parsing the json and that too in a fail-safe manner.

Reply:
fixed as per the commend of Rajendra.